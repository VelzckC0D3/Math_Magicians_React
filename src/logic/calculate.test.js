import calculate from './calculate';

describe('calculate function', () => {
  it('should return an object with null values when buttonName is AC', () => {
    const obj = { total: '10', next: '5', operation: '+' };
    const buttonName = 'AC';
    const result = calculate(obj, buttonName);
    expect(result).toMatchSnapshot();
  });

  it('should update the next value when buttonName is a number', () => {
    const obj = { total: '10', next: '5', operation: '+' };
    const buttonName = '3';
    const result = calculate(obj, buttonName);
    expect(result).toMatchSnapshot();
  });

  it('should update the next value to "0." when buttonName is "." and there is an operation but no next value', () => {
    const obj = { total: '10', operation: '+' };
    const buttonName = '.';
    const result = calculate(obj, buttonName);
    expect(result).toMatchSnapshot();
  });

  it('should perform the operation and return the result when buttonName is "="', () => {
    const obj = { total: '10', next: '5', operation: '+' };
    const buttonName = '=';
    const result = calculate(obj, buttonName);
    expect(result).toMatchSnapshot();
  });
});
