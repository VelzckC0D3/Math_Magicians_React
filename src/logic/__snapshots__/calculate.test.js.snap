// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculate function should perform the operation and return the result when buttonName is "=" 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "15",
}
`;

exports[`calculate function should return an object with null values when buttonName is AC 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": null,
}
`;

exports[`calculate function should update the next value to "0." when buttonName is "." and there is an operation but no next value 1`] = `
Object {
  "next": "0.",
  "operation": "+",
  "total": "10",
}
`;

exports[`calculate function should update the next value when buttonName is a number 1`] = `
Object {
  "next": "53",
  "operation": "+",
  "total": "10",
}
`;
